drivers/hmis/app/models/hmis/hoh_change_handler.rb

```diff
@@ -0,0 +1,113 @@
+module Hmis
+  #
+  # The HohChangeHandler class is responsible for handling changes to the Head of Household (HoH) in a household enrollment.
+  # It provides methods to perform the change, apply the changes to the database, and validate the changes.
+  #
+  # This logic is split out from the UpdateRelationshipToHoH mutation in case we need to use it in cleanup routines,
+  # for example bulk assigning HoH to the oldest adult in a household.
+  #
+  # Parameters:
+  # - new_hoh_enrollment: The Enrollment that is becoming the Head of Household
+  # - hud_user_id: The "UserID" of the HUD user making the change.
+  class HohChangeHandler
+    attr_accessor :new_hoh_enrollment, :household_enrollments, :old_hoh_enrollments, :new_hoh_move_in_date, :old_hoh_move_in_date, :validation_errors, :hud_user_id
+
+    def initialize(new_hoh_enrollment:, hud_user_id:)
+      self.new_hoh_enrollment = new_hoh_enrollment
+      self.hud_user_id = hud_user_id
+
+      # All Enrollments in the household (including new and old HoHs)
+      self.household_enrollments = new_hoh_enrollment.household_members.preload(:client)
+      # Previous HoHs (should only be one, but we handle multiple for data cleanup)
+      self.old_hoh_enrollments = household_enrollments.sort_by(&:DateCreated).select(&:head_of_household?)
```

Comment:
I think there should be a tie breaker, using id. Also I prefer `filter` rather than `select` with arrays to disambiguate from active_record's `select` method
```suggestion
      self.old_hoh_enrollments = household_enrollments.
        filter(&:head_of_household?).
        sort_by { |e| [e.DateCreated, e.id] }
```

Reply:
updated, thanks!